### PR TITLE
[TACHYON-1627] Refactor current documentation framework to support multiple languages

### PR DESCRIPTION
**Pull request by saltylin**
https://tachyon.atlassian.net/browse/TACHYON-1627

The file structure and the documentation framework  in docs/ have be refactored so that it can support multiple languages website further.

### CODE REVIEW

**Comment by saltylin** [comment]
Next step we will translate all pages into Chinese, and language-independent resources (e.g. code fragments and configuration properties) will be extract into public folders at the same time.

File: docs/_layouts/global.html [hunk 1]
```diff
@@ -87,8 +89,8 @@
                                         </a>
                                     </li>
                                 {% else %}
-                                    <!-- trim the leading slash to make the path relative to preserve version links -->
-                                    <li><a href="{{ node.url | remove-first: '/' }}">
+                                    <!-- Use the prefix .. to change the absulote url to relative url -->
+                                    <li><a href="..{{ node.url }}">
```
Comment (calvinjia) [inline docs/_layouts/global.html]: Is it guaranteed that `node.url` will have a leading slash?

**Comment by calvinjia** [comment]
Jenkins, add to whitelist.

**Comment by calvinjia** [comment]
Jenkins, ok to test

**Comment by calvinjia** [comment]
@saltylin Thanks for the improvement, could you provide a screenshot of the before and after for the docs page?

**Comment by saltylin** [comment]
@calvinjia I've create a github project page here (http://saltylin.github.io/tachyon-site-demo) to show the effects after modification.
This is what it looks like before modification below.
before
And the English page after modification.
after-english
And Chinese page.
after-chinese
The cn/ folder should be empty since I haven't translated any page now, so I just copy index.md from en/ to cn/ and simply change its title to "首页". Thus in Chinese page view all the dropdown menus is empty.
I have modify the main.css by adding `padding-right: 5px;` in `.navbar .nav>li>a{...}` statement. Since a new language selector dropdown menu is here and the width of navigation bar becomes larger, if I don't add it, the whole navigation bar will shift down a line.
In Chinese page view the navigation items seem to be a bit narrow. Should I add some css statements to make them looks wider?

File: docs/_layouts/global.html [hunk 2]
```diff
@@ -117,7 +131,7 @@
 
 <div class="container" id="content">
     <h1 class="title">{{ page.title }}</h1>
-
+
```
Comment (apc999) [inline docs/_layouts/global.html]: unnecessary whitespace here?

File: docs/_layouts/global.html (hunk 1 above)
Comment (apc999) [inline docs/_layouts/global.html]: question: is this prefix of `..` in  `..{{ ` intentional?

Comment (apc999) [inline docs/_layouts/global.html]: saw the comments,

**Comment by apc999** [comment]
+1 LGTM, thanks @saltylin for the awesome improvement!

File: docs/_layouts/global.html (hunk 2 above)
Comment (saltylin) [inline docs/_layouts/global.html]: Sorry, I have removed it.

File: docs/_layouts/global.html (hunk 1 above)
Comment (saltylin) [inline docs/_layouts/global.html]: Yes. Since the documentation site isn't on the root path of the website, absolute path will cause error. But this approach means all page files must be in en/ (or cn/) directory exactly and no subdirectory could be created to contain pages. I'm also wondering is there a better approach for this case.

**Comment by saltylin** [comment]
@calvinjia The conflicts are fixed.

**Comment by gpang** [comment]
ok to test

**Comment by apc999** [comment]
+1 LGTM
